packages/frontend/src/components/RichTextEditor/MenuBar.tsx

```diff
@@ -190,35 +224,147 @@ interface MenuBarProps {
   editor: Editor | null
 }
 export const MenuBar = ({ editor }: MenuBarProps) => {
+  const [showValueDialog, setShowValueDialog] = useState(false)
+  const [dialogValue, setDialogValue] = useState('')
+  const [dialogLabel, setDialogLabel] = useState<MenuLabels | null>(null)
+  const onClickOverrides: { [key: string]: () => void } = {
+    [MenuLabels.LinkSet]: useCallback(() => {
+      if (!editor) {
+        return
+      }
+      const previousUrl = editor.getAttributes('link').href
+      setDialogLabel(MenuLabels.LinkSet)
+      setDialogValue(previousUrl)
+      setShowValueDialog(true)
+    }, [editor]),
+    [MenuLabels.ImageAdd]: useCallback(() => {
+      if (!editor) {
+        return
+      }
+      setDialogLabel(MenuLabels.ImageAdd)
+      setDialogValue('')
+      setShowValueDialog(true)
+    }, [editor]),
+  }
+  const dialogOnSubmits: { [key: string]: () => void } = {
+    [MenuLabels.LinkSet]: useCallback(() => {
+      if (!editor) {
+        return
+      }
+      const url = dialogValue
+      // cancelled
+      if (url === null) {
+        return
+      }
+
+      // empty
+      if (url === '') {
+        editor.chain().focus().extendMarkRange('link').unsetLink().run()
+
+        return
+      }
+
+      // update link
+      editor
+        .chain()
+        .focus()
+        .extendMarkRange('link')
+        .setLink({ href: url })
+        .run()
+      setShowValueDialog(false)
+    }, [editor, dialogValue, setShowValueDialog]),
+    [MenuLabels.ImageAdd]: useCallback(() => {
+      if (!editor) {
+        return
+      }
+      const url = dialogValue
+      if (url === null) {
+        return
+      }
+      editor.chain().focus().setImage({ src: url }).run()
+      setShowValueDialog(false)
+    }, [editor, dialogValue]),
+  }
+  const dialogPlaceholders: { [key: string]: string } = {
+    [MenuLabels.LinkSet]: 'Enter a full URL with http prefix',
+    [MenuLabels.ImageAdd]:
+      'Enter direct image link (e.g. https://file.go.gov.sg/clipplumer.png)',
```

Comment:
nit: typo here? or without the letter b on purpose hmmmm

Reply:
lol ok i go fix the link. It's an actual link I typo-ed